### PR TITLE
Display notices on cart/checkout block templates

### LINKED ISSUE #10113: Notice doesn't appear on the cart block page if the product is added via a direct link in a block-based theme
## Describe the bug

Notice doesn't appear on the cart block page if the product is added via a direct link in a block-based theme. 

## To reproduce

1. Activate block based theme like Twenty-Twenty-three
2. Add product via add to cart link. (Eg: http://localhost:10013/cart/?add-to-cart=34)

## Expected behavior

The product has been added to the cart successfully notice should appear. 

## Screenshots

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/de992ce9-5d84-4df2-9f1b-0e3714422d91

## Additional context

Slack: p1688630512127629-slack-C02UBB1EPEF

### PR DESCRIPTION
**Pull request by wavvves**
The notice for adding a new item via `?add-to-cart=xx` was not displaying on Cart and Checkout templates.

Fixes #10113

#### User Facing Testing

1. Activate a based theme like TT3
2. Go to Appearance > Editor > Templates > Manage All Templates
3. Click the three dots next to the Cart and Checkout templates and then clear customizations
4. Add product via add to cart link. (Eg: http://localhost/cart/?add-to-cart=34)
5. Verify that the notice stating that the item was added to the cart appears

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [X] Feature plugin
* [ ] Experimental

### Changelog

> Fixed notice displaying on Cart and Checkout template

### CODE REVIEW

**Review by tarhi-saad — COMMENTED** [review]
The PR works as expected! I still need to check the code, but I think @opr will handle it!

File: templates/templates/blockified/cart.html [hunk 1]
```diff
@@ -148,7 +149,7 @@
 				<!-- wp:pattern {"slug":"woocommerce/cart-empty-message"} /-->
 
 				<!-- wp:separator {"className":"is-style-dots"} -->
-				<hr class="wp-block-separator has-alpha-channel-opacity is-style-dots" />
```
Comment (opr) [inline templates/templates/blockified/cart.html]: Sorry this is nitpicky but wondering why the space was removed. Not going to block the PR but wondering if it was intentional, and if so, based on what?

**Review by tarhi-saad — APPROVED** [review]
I checked the code, @wavvves! It looks good! I believe you can 🚢 it

**Review by opr — DISMISSED** [review]
I had to revert the template to get this to work.

Adding to cart by the link `site.local/cart/?add-to-cart=18` did show a notice after reverting.

Is there a way to get this fix in without requiring the user to revert their customisations?

Also, one non-blocking question about code style.

File: templates/templates/blockified/cart.html (hunk 1 above)
Comment (wavvves) [reply templates/templates/blockified/cart.html]: It wasn't intentional, probably autoformatting. I'll revert that

**Comment by wavvves** [comment]
> Re-tested and not working

Can we sync up? Maybe you have some testing scenario I did not account for, I cleaned everything and re-tested both with migrated content, and clean template, and both showed the notifications for that add to cart flow.

maybe try this

1. Switch to a non-block theme and make sure the notice appears on your cart page
2. Delete the has_migrated_cart and has_migrated_checkout options, switch to a block theme, and re-test it for migrated content
3. Delete customizations and re-test it again

**Comment by thealexandrelara** [comment]
I'm bumping the milestone for this PR to 10.8.0.

**Comment by wavvves** [comment]
I'm proceeding with this update from @opr:

> So basically, the flow is this:
> 
> 1. Check out trunk
> 2. Clear customisations on your Cart template
> 3. Check out the PR
> 4. It doesn't work

**Comment by imanish003** [comment]
I am going to bump this PR for the next release as it doesn't have a blocker label.

**Comment by wavvves** [comment]
> > Re-tested and not working
> 
> Can we sync up? Maybe you have some testing scenario I did not account for, I cleaned everything and re-tested both with migrated content, and clean template, and both showed the notifications for that add to cart flow.
> 
> maybe try this
> 
> 1. Switch to a non-block theme and make sure the notice appears on your cart page
> 2. Delete the has_migrated_cart and has_migrated_checkout options, switch to a block theme, and re-test it for migrated content
> 3. Delete customizations and re-test it again

I still can't replicate this.

1.  If content was migrated (either shortcode or blocks) -> notices work
2. Default template (cleared customizations) -> notices work
3. Default template on trunk -> notices don't work (as expected)
4. Default template after switching into this branch (equivalent to updating blocks) -> notices work

If there's no content saved it will always fetch from template boilerplate code. Have you tried hard refresh or incognito?

**Review by opr — APPROVED** [review]
Works for me now after clearing customizations on my template!

**Comment by opr** [comment]
Let's fix the `WooCommerce Visibility` section and add the cart & blocks labels for this PR @wavvves

**Pull request merged**